test/extensions/filters/http/wasm/wasm_filter_test.cc

```diff
@@ -1436,7 +1436,8 @@ TEST_P(WasmHttpFilterTest, RootId2) {
   setupFilter("context2");
   EXPECT_CALL(filter(), log_(spdlog::level::debug, Eq(absl::string_view("onRequestHeaders2 2"))));
   Http::TestRequestHeaderMapImpl request_headers;
-  EXPECT_EQ(Http::FilterHeadersStatus::Continue, filter().decodeHeaders(request_headers, true));
+  EXPECT_EQ(Http::FilterHeadersStatus::StopAllIterationAndWatermark,
+            filter().decodeHeaders(request_headers, true));
```

Comment:
Given that the filter code is in a different repo, does it make sense to have assertions like this one about the return values of decodeHeaders in the main repo?

Reply:
The test filter is in this repo, this is the test case for which the behavior changed:
https://github.com/envoyproxy/envoy/blob/1d6d68bd659a78ad53b46799efa2f810c3eb2dd7/test/extensions/filters/http/wasm/test_data/test_cpp.cc#L750-L754

Reply:
So the coding error was in a wasm filter implementation, not in the wasm framework itself?